### DISCUSSION TITLE
Passage de souris + affichage scène

### DISCUSSION THREAD

**Discussion by Baptiste-ALL** [witsa/synapps#742] [Q&A]
Bonjour, tous d'abord merci à vous pour ce logiciel.
je souhaite trouvé le moyen lorsqu'on survol un bouton avec la souris, de faire affiché une scène dans un petit écran. 

Merci d'avance

**Comment by WITNicolas** [comment]
Bonjour,

Il faudra une modale (qui sera le popup) et il y a aura besoin de créer 2 évènements sur le bouton.

OnMouseOver
context.getActor('CLE DE LA MODALE').properties.isShown = true

OnMouseOut 
context.getActor('CLE DE LA MODALE').properties.isShown = false

![image](https://github.com/witsa/synapps/assets/33632664/2541b829-fb2b-4f13-b74b-a8fd6ed2fc8c)

**Comment by Baptiste-ALL** [comment]
Merci pour la réponse rapide j'ai donc essayé mais coté (popup) cela m'affiche 

error during onmouseover with 'cle du bouton'

et j'aimerais affiché une autre scène dans ce popup différente que celle du bouton.

**Comment by Raphwit** [comment]
Bonjour, voici une scène sommaire pour vous inspirer pour le comportement que vous souhaitez : 

```
SYNAPPS-STUDIO-SCENE|{"config":{"key":"scene1","name":"Scène 1"},"leadActor":{"type":"layout/stack","key":"stack1","children":[{"type":"input/button","key":"button1","properties":{"content":"Button","mode":"primary"},"events":{"onMouseOver":["context.getActor('modal1').properties.isShown = true;"],"onMouseOut":["context.getActor('modal1').properties.isShown = false;"]}},{"type":"layout/modal","key":"modal1","properties":{"height":"200px","width":"200px","translucentOverlay":false},"children":[{"type":"layout/stack","key":"stack2","properties":{"verticalAlignment":"expand"},"children":[{"type":"display/text","key":"text1","properties":{"content":"HELLO WORLD","horizontalAlignment":"middle","verticalAlignment":"middle"}}]}]}]}}
```

Il suffit de coller cette scène dans Synapps pour l'obtenir, elle contient un bouton qui affiche une modale lorsque l'on passe la souris dessus !

Attention, si la modale ne n'as pas un overlay transparent, cela ne fonctionne pas !

**Reply by MickaGaffet** [reply]
Bonjour, j'aimerais garder des info au passage de la souris, mais également au click afficher une "pop up" qui par soit sur un bouton "fermé" soit en cliquant en dehors de la "pop up" 
L'affichage j'y arrive a peut prêt, mais pour la fermeture c'est plus compliquée

**Reply by Raphwit** [reply]
Bonjour, vous pouvez utiliser la variante 'modale au click' de l'acteur modale afin de sélectionner un acteur qui permet l'ouverture et un acteur qui permet la fermeture.

**Comment by Baptiste-ALL** [comment]
Merci beaucoup cela me permet d'avance. Justement j'ai fait un essai sa affiche bien le POPUP et le mot HELLOWORLD que j'ai voulus remplacé part ENERGIE {{gaz}}  et dans gaz je suis allé cherche une ressource.
au final sa affiche ENERGIE :easy.RESS.R ....  j'ai essayé pas mal de truc diffèrent mais je bloque sinon c'était un texte et un autre texte avec la valeur mais j'aimerais faire plus propre. 
Merci d'avance

**Comment by Raphwit** [comment]
Lors de la liaison de cette ressource, il faut préciser le champ à aller chercher, c'est probablement ce qu'il vous manque ! 

Voici l'exemple sur une liaison vers une ressource : 
![image](https://github.com/witsa/synapps/assets/93767525/721d8d53-675c-4e63-bff5-93c2789971eb)

**Comment by Baptiste-ALL** [comment]
ok c'est bon c'est que le champ n'apparaisse pas car je n'avais pas lier la ressource 
j'avais juste sélectionné la ressource merci
